L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -133,7 +133,7 @@ float HGCalShowerShape::percentileLayer(const l1t::HGCalMulticluster& c3d,
   }
   // Linear interpolation of percentile value
   double pt0 = (percentile > 0 ? layers[percentile - 1] : 0.);
-  double pt1 = layers[percentile];
+  double pt1 = (percentile < layers.size() ? layers[percentile] : layers.back());
```

Comment:
Can `layers.size() == 0`? If so then calling `layers.back()` in that case reads off the end (and may cause a segmentation fault since it is likely to be reading memory location 0).

Reply:
The size of `layers` is given by the number of HGCAL layers :
https://github.com/cms-sw/cmssw/blob/04f9376ae2a8d215fb8f906b71e8ab5b2517caa8/L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc#L111-L112
where the trigger tools returns this:
https://github.com/cms-sw/cmssw/blob/04f9376ae2a8d215fb8f906b71e8ab5b2517caa8/L1Trigger/L1THGCal/src/HGCalTriggerTools.cc#L72-L73
Which is computed from the HGCAL geometry:
https://github.com/cms-sw/cmssw/blob/04f9376ae2a8d215fb8f906b71e8ab5b2517caa8/L1Trigger/L1THGCal/src/HGCalTriggerTools.cc#L38-L47

So unless something really bad happens it shouldn't be 0